File: 프로그래머스/unrated/154538. 숫자 변환하기/숫자 변환하기.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int x, int y, int n) {
        int answer = -1;
        Queue<int[]> q = new LinkedList<>();
        q.add(new int[]{y,0});

        while(!q.isEmpty()){
            int[] now = q.poll();
            if(now[0] == x){
                return now[1];
            }
            if(now[0] > x){
                if(now[0] % 3 == 0){
                    q.add(new int[]{now[0]/3,now[1] + 1});
                }
                if(now[0] % 2 == 0){
                    q.add(new int[]{now[0]/2,now[1] + 1});
                }
                q.add(new int[]{now[0] - n,now[1] + 1});
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.py
from collections import deque

def solution(x, y, n):
    answer = -1
    q = deque([(y,0)])
    
    while q:
        # 큐에서 현재 값과 연산 횟수를 꺼냄
        now, count = q.popleft()

        # 목표 값에 도달한 경우
        if now == x:
            return count

        # now가 x보다 크면 연산을 진행
        if now > x:
            # 3으로 나누어 떨어지면 3으로 나눈 값을 큐에 추가
            if now % 3 == 0:
                q.append((now // 3, count + 1))
            # 2로 나누어 떨어지면 2로 나눈 값을 큐에 추가
            if now % 2 == 0:
                q.append((now // 2, count + 1))
            # n을 빼면 새로운 값을 큐에 추가
            q.append((now - n, count + 1))
            
    return answer